impl/maven-impl/src/test/java/org/apache/maven/impl/model/DefaultModelProcessorTest.java

```diff
@@ -0,0 +1,184 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.impl.model;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.maven.api.model.Model;
+import org.apache.maven.api.services.Source;
+import org.apache.maven.api.services.xml.ModelXmlFactory;
+import org.apache.maven.api.services.xml.XmlReaderRequest;
+import org.apache.maven.api.spi.ModelParser;
+import org.apache.maven.api.spi.ModelParserException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class DefaultModelProcessorTest {
+
+    @Test
+    void readWithValidParserShouldReturnModel() throws Exception {
+        ModelXmlFactory factory = mock(ModelXmlFactory.class);
+        ModelParser parser = mock(ModelParser.class);
+        XmlReaderRequest request = mock(XmlReaderRequest.class);
+        Model model = mock(Model.class);
+        Path path = Path.of("project/pom.xml");
+
+        when(request.getPath()).thenReturn(path);
+        when(request.isStrict()).thenReturn(true);
+        when(model.withPomFile(path)).thenReturn(model);
+        when(parser.locateAndParse(any(), any())).thenReturn(Optional.of(model));
+
+        DefaultModelProcessor processor = new DefaultModelProcessor(factory, List.of(parser));
+        Model result = processor.read(request);
+
+        assertNotNull(result);
+        assertEquals(model, result);
+    }
+
+    @Test
+    void readAllParsersFailShouldFallbackToFactory() throws Exception {
+        ModelXmlFactory factory = mock(ModelXmlFactory.class);
+        ModelParser parser = mock(ModelParser.class);
+        XmlReaderRequest request = mock(XmlReaderRequest.class);
+        Model fallbackModel = mock(Model.class);
+        Path path = Path.of("project/pom.xml");
+
+        when(request.getPath()).thenReturn(path);
+        when(request.isStrict()).thenReturn(false);
+        when(parser.locateAndParse(any(), any())).thenThrow(new ModelParserException("parse fail"));
+
+        when(factory.read(request)).thenReturn(fallbackModel);
```

Comment:
This mock here seems to make the test meaningless. It looks like all you're testing is what the mock returns when the mocked method is called.

Reply:
good eye. yes its boilerplate:

<img width="1824" alt="image" src="https://github.com/user-attachments/assets/d37df953-1e83-4c00-9d27-a80d2f510652" />

Reply:
gone